import { AppError } from "app/helpers/app-error";
import * as z from "zod";

export const Schema = z.object({
  // name: z.string().min(1).max(255),
});

export type SchemaType = z.infer<typeof Schema>;

export async function {{{ camelCase(name) }}}(data: SchemaType) {
  try {
    Schema.parse(data);
  } catch (error) {
    if (error instanceof z.ZodError || error instanceof AppError) {
      throw error;
    }

    throw AppError.E_GENERIC_ERROR("Unexpected error during validation.");
  }
}
